config/types/directory.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"path/filepath"
+)
+
+// Just a special case of File
+type Directory File
+
+func (d *Directory) NumSegments() int {
+	p := filepath.Clean(string(d.Path))
+	count := 0
+	for p != "/" {
+		count++
+		p = filepath.Dir(p)
+	}
+	return count
+}
+
+// Needed to sort directories so /foo get created before /foo/bar if they are both specified
+type ByPathSections []Directory
```

Comment:
The NumSegments needs to stay in this package if it is to stay a method since you can't define methods on types outside the current package, unless you make it a function that takes the Directory as a parameter instead or make a type alias. The ByDirectorySegments can be moved out though. Which were you referring to?